### PR TITLE
fix: user profile not deleted without "Manage User Access" permission gf-449

### PR DESCRIPTION
**Pull request by Vitaliistf**
Added logic to handle user self-accessing activity to the check user permissions hook.

### CODE REVIEW

File: apps/backend/src/libs/hooks/check-user-permissions.hook.ts [hunk 1]
```diff
@@ -16,6 +16,14 @@ const checkUserPermissions = (routePermissions: string[]): APIPreHandler => {
 			});
 		}
 
+		const resourceId = (params as { id?: string }).id;
+
+		if (url.includes(APIPath.USERS) && Number(resourceId) === user.id) {
```
Comment (perekotypole) [inline apps/backend/src/libs/hooks/check-user-permissions.hook.ts:21]: How about adding a variable like `isCurrentUserRoute`,  `hasFreeAccess` or something for the condition? Just to keep semantic

Comment (Vitaliistf) [reply apps/backend/src/libs/hooks/check-user-permissions.hook.ts]: Thanks, nice notice. I've changed this, however decided on a little different names.

File: apps/backend/src/libs/hooks/check-user-permissions.hook.ts [hunk 2]
```diff
@@ -16,6 +16,16 @@ const checkUserPermissions = (routePermissions: string[]): APIPreHandler => {
...
 
+		const resourceId = (params as { id?: string }).id;
+		const isUsersRoute = url.includes(APIPath.USERS);
+		const isCurrentUserRoute = isUsersRoute && Number(resourceId) === user.id;
+
+		if (isCurrentUserRoute) {
+			done();
+
+			return;
+		}
```
Comment (liza-veis) [inline apps/backend/src/libs/hooks/check-user-permissions.hook.ts:27]: This logic shouldn't be in checkUserPermissions hook. This hook is for common permission case and should now nothing about specific routes and their logic. Instead let's use DELETE DELETE /authenticated-user and PUT /authenticated-user in auth controller for current user, and move the logic related to this routes to auth module

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: apps/backend/src/libs/hooks/check-user-permissions.hook.ts (hunk 2 above)
Comment (liza-veis) [reply apps/backend/src/libs/hooks/check-user-permissions.hook.ts:27]: yes

**Review by liza-veis — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**